### PR TITLE
feat: speed up MockFileSystem constructor

### PR DESCRIPTION
**Pull request by schibu007**
This PR is an alternative to the PR #695. It has a slightly better performance and does not change the concept of having all paths in one collection.
I created this PR after discussion with @jflepp.

### CODE REVIEW

File: tests/System.IO.Abstractions.TestingHelpers.Tests/MockFileSystemTests.cs [hunk 1]
```diff
@@ -351,6 +352,22 @@ public void MockFileSystem_Constructor_ThrowsForNonRootedCurrentDirectory()
...
+            var maxDirectoryDepth = 8;
+            var testData = Enumerable.Range(0, filesCount).ToDictionary(
+                i => XFS.Path(@$"C:\{string.Join(@"\", Enumerable.Range(0, i % maxDirectoryDepth + 1).Select(i => i.ToString()))}\{i}.bin"),
+                i => new MockFileData(i.ToString()));
+            var stopWatch = Stopwatch.StartNew();
+
+            var mockFileSystem = new MockFileSystem(testData);
+
+            stopWatch.Stop();
+            Assert.IsTrue(stopWatch.Elapsed < TimeSpan.FromSeconds(30));
```
Comment (fgreinacher) [inline tests/System.IO.Abstractions.TestingHelpers.Tests/MockFileSystemTests.cs:368]: Great that you wrote a test to verify your changes! 

I prefer to keep the tests on the functional level and very fast. Would you mind moving this test case to the `benchmarks` directory to accommodate for this?

**Comment by fgreinacher** [comment]
Thanks @schibu007 (and @jflepp) for your contribution 🤩 I prefer this version because it changes much less and I don't see any drawbacks here. I left one minor comment and will merge once this is addressed.

File: tests/System.IO.Abstractions.TestingHelpers.Tests/MockFileSystemTests.cs (hunk 1 above)
Comment (schibu007) [reply tests/System.IO.Abstractions.TestingHelpers.Tests/MockFileSystemTests.cs]: I moved the test case to the `benchmarks` project. I had to reference `System.IO.Abstractions.TestingHelpers.csproj`.

**Comment by fgreinacher** [comment]
Looks great. Thanks a lot @schibu007 and @jflepp 😉

**Pull request merged**